### PR TITLE
Stopped announcing the AlertDialog's route since it is redundant

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

The first item on the alert has the same text as the route, so it is redundant.

With my testing Android behaves exactly the same, on iOS the alert's name isn't announced twice.

## Related Issues

https://github.com/flutter/flutter/issues/44001

## Tests

Manual testing, there isn't integration tests for the accessibility code.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -490,7 +490,7 @@ class AlertDialog extends StatelessWidget {
           style: titleTextStyle ?? dialogTheme.titleTextStyle ?? theme.textTheme.headline6,
           child: Semantics(
             child: title,
-            namesRoute: label == null,
+            namesRoute: false, // No need to announce the route since the title of the alert has the same text.
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/dialog.dart:493]: We can just remove this line.

**Review by chunhtai — COMMENTED** [review]
We need the a test for this change.

**Comment by chunhtai** [comment]
Hi @gaaclarke , Are you still working on this PR?

**Comment by gaaclarke** [comment]
No.  After further consideration, I think the current behavior is correct after having played with iOS and Android.  On Android it is announced that an alert is presented, on iOS it doesn't.  If we removed the announcement of the route, all semantics hints of its presence will be removed.  Yes, the title of the alert and the route are the same, so it will be spoken twice.  When coming back to a dialog it would be equally valid for voiceover to select the last selected item which wouldn't be the title.  Then not speaking the route would be even more confusing.  If someone wants a different behavior I think they could get it by creating their own alert.  It wouldn't be correct to have that alert in the framework though.

**Comment by chunhtai** [comment]
Will the iOS focus anything other than the tittle when a new dialog is open?

**Comment by gaaclarke** [comment]
@chunhtai the way we've currently implemented things on ios it will always select the title.  On android it selects the last selected item.

**Comment by chunhtai** [comment]
For screen changes, we now only passes in the string, we do now force iOS to pick a title, it is by default that iOS do that.
However, if that is the case,  aren't we always sure the alert title will be announce twice for iOS? in that case, should we remove the name route, or give it an empty string when the platform is iOS?

**Comment by gaaclarke** [comment]
This is the correct change if you want to get rid of it I think.  I'm of the opinion that that isn't necessarily the correct thing to do.

**Comment by chunhtai** [comment]
ok, I just remembered I made a change to be able to add alert semantics label without introducing an additional node, if developer does not want this behavior they can give it an empty string.
```dart
AlertDialog(
 SemanticsLabel:  " "
)
```

So I think this is fine.

**Pull request closed without merging**